src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/EventGridExtensions.cs

```diff
@@ -0,0 +1,59 @@
+using Microsoft.Azure.EventGrid.Models;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+	/// <summary>
+	/// 
+	/// </summary>
+	public static class EventGridEventExtensions
+    {
+		private static string dataSyncToken = "syncToken";
+
+		/// <summary>
+		/// This method uses an EventGridEvent from EventGrid and tries to create a <see cref="PushNotification"/>
+		/// </summary>
+		/// <param name="eventGridEvent"> EventGridEvent from EventGrid</param>
+		/// <param name="pushNotification"> out parameter set up in this method</param>
+		/// <returns></returns>
+		public static bool TryCreatePushNotification(this EventGridEvent eventGridEvent, out PushNotification pushNotification)
+		{
+			pushNotification = null;
+
+			if (eventGridEvent == null || eventGridEvent.Data == null || eventGridEvent.EventType == null || eventGridEvent.Subject == null)
+			{
+				return false;
+			}
+
+			try
+			{
+				string eventGridData = eventGridEvent.Data.ToString();
+				JsonElement jsonMessage = JsonDocument.Parse(eventGridData).RootElement;
```

Comment:
These two statements can be combined into:
`JsonElement jsonMessage = JsonDocument.Parse(eventGridEvent.Data.ToString()).RootElement;`

Reply:
Actually, we can combine all 3:

`string syncToken = JsonDocument.Parse(eventGridEvent.Data.ToString()).RootElement.GetProperty(dataSyncToken).GetString();`